coralogix/terraform-coralogix-aws
# Feature: OTEL service daemon for ECS on EC2 container instances.

# Description

Terraform module provisioning OTEL agent collector as an ECS service daemon. Supports EC2 container instances.

Feature request: [CDS-762 ](https://coralogix.atlassian.net/browse/CDS-762)Create OTEL ecs-ec2 Terraform.

# How Has This Been Tested?

* Environment Setup:
  * Create New AWS VPC, ECS Cluster on EC2 container instances. 
  * Deploy test nginx service.
  * Open a terminal with valid AWS session credentials and Coralogix key as ``` TF_VAR_private_key```
  * Review ```terraform-coralogix-aws/tests/ecs-ec2/ecs-ec2.tf```, edit ```region``` and ```coralogix_region``` accordingly to your environment.
  * Deploy ```ecs-ec2``` Terraform module:
    ```terraform
    cd terraform-coralogix-aws/tests/ecs-ec2
    terraform init
    terraform plan
    terraform apply -auto-approve
    ```
* Testing: 
  * Verify ECS service named ```coralogix-otel-agent``` runs as a daemon task on all EC2 instances.
  * Send requests to nginx container endpoint.
  * Verify Coralogix console shows nginx logs and container metrics

# Checklist:
- [ ✅ ] I have updated the relevant example in the examples directory for the module.
- [ ✅ ] I have updated the relevant test file for the module.
- [ ✅ ] This change does not affect module (e.g. it's readme file change)
